glass/src/lib/native/include/glass/support/ExpressionParser.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <stdint.h>
+
+namespace glass::expression {
+
+enum class ParseResultKind {
+  Success,
+  Error,
+};
+
+template <typename V>
+struct ParseResult {
+  static ParseResult<V> CreateSuccess(V value);
+  static ParseResult<V> CreateError(const char* errorMessage);
```

Comment:
I'd feel better if either these declarations were moved to the `.cpp` file (They would probably have to be changed to free functions) or these were defined in the `.h` file, since otherwise someone might try to use these functions in a different `.cpp` file and won't be able to instantiate the template.

Reply:
I'm currently unsure as to whether `ParseResult` will stay since ideally it would use `wpi::expected`, but I haven't done that yet due to the conflict I described above [here](https://github.com/wpilibsuite/allwpilib/pull/6530#discussion_r1577106243). If it does end up staying I will do that :+1: